pkg/gameserverallocations/allocation_cache.go

```diff
@@ -70,6 +70,9 @@ func NewAllocationCache(informer informerv1.GameServerInformer, counter *gameser
 			if !ok {
 				return
 			}
+			if oldGs.ObjectMeta.ResourceVersion == newGs.ObjectMeta.ResourceVersion {
```

Comment:
I like it! Good find!

Worth a unit test, or nah?

Reply:
Oh, I will add an unit test.